### ISSUE TITLE
Custom loss with VGG16 features works in keras and doesn't work in tf.keras

### ISSUE THREAD

**Issue by muxgt** [tensorflow/tensorflow#34595]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Google Colab. Ubuntu 18.04.3
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary):
- TensorFlow version (use command below): 1.15
- Python version: 3
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**

I'm using tensorflow.keras with TF 1.15 in Colab. I have custom loss function that loads VGG16 from tf.keras and does feature extraction from y_pred and y_true. This feature extraction works if I pass as arguments either y_true or y_pred or any expression without y_true. If I pass expression with y_true and something else it raises error: `You must feed a value for placeholder tensor 'dense_4_target' with dtype float and shape [?,?,?,?]
	 [[{{node dense_4_target}}]]` The same code works when I change tensorflow.keras to just keras.

**Describe the expected behavior**

Model should compile without errors as in pure Keras

**Code to reproduce the issue**
Please find gist [here](https://colab.research.google.com/gist/muxgt/c467f0a999f3a517264040465af4a5a0/untitled10.ipynb#scrollTo=GEBbaquGImyU)

**Comment by muxgt** [comment]
Dear Team, any news on this?

**Comment by jvishnuvardhan** [comment]
@muxgt Looks like we need to supply a numpy array or eager tensor for `extra_input`.
When I changed `extra_input = Input((256, 256, 3))` to `extra_input = np.ones((256,256,3))`, it worked as expected. Thanks!

As error mentioned, you could create a placeholder and feed the value for the placeholder.

Feel free to close the issue if this was resolved for you. Thanks!

**Comment by jvishnuvardhan** [comment]
@muxgt I am closing this issue as it was resolved. Please feel free to reopen if the issue persists again. Thanks!

**Comment by muhongqiang** [comment]
TD



在 2020-03-27 09:45:13，"Vishnuvardhan Janapati" <notifications@github.com> 写道：

Closed #34595.

**Comment by muhongqiang** [comment]
unsubscribe



At 2020-03-27 09:43:11, "Vishnuvardhan Janapati" <notifications@github.com> wrote:

@muxgt I am closing this issue as it was resolved. Please feel free to reopen if the issue persists again. Thanks!
